### DISCUSSION TITLE
Wrong behaviour configuring stateful reconnect

### DISCUSSION THREAD

**Discussion by bdovaz** [dotnet/aspnetcore#55996] [General]
https://learn.microsoft.com/en-us/aspnet/core/signalr/configuration?view=aspnetcore-8.0&tabs=dotnet#configure-stateful-reconnect

@davidfowl I have the same problem as @quytrieu37 in #54818 and I see he did not manage to solve it....

I mean, between reconnections I get different connection ids.... The client is configured with `WithAutomaticReconnect` and `WithStatefulReconnect`. In the documentation it doesn't say anything about not setting `WithAutomaticReconnect` and I have removed it but when I do that what happens is that the HubConnection goes directly to the disconnected state...

Is there any way to debug this problem to know if it is a problem in the client or in the server? Even if I set the logs in client and server to `Trace` I don't see anything useful beyond the negotiation url that adds the `withstatefulreconnect=true` (or something similar).

Thanks.

**Comment by davidfowl** [comment]
cc @BrennanConroy

**Comment by co-dax** [comment]
See [https://github.com/dotnet/aspnetcore/discussions/54818#discussioncomment-10996819](https://github.com/dotnet/aspnetcore/discussions/54818#discussioncomment-10996819) for a possible workaround.

**Comment by Mise22** [comment]
I have the same issue. I can confirm `WithAutomaticReconnect` is needed. In the source code I can't find where the `ConnectionId` preservation is supposed to happen.
(full reply https://github.com/dotnet/aspnetcore/discussions/54818#discussioncomment-11121458)
